extensions/cornerstone-dicom-sr/src/viewports/OHIFCornerstoneSRViewport.tsx

```diff
@@ -69,18 +71,25 @@ function OHIFCornerstoneSRViewport(props) {
     MEASUREMENT_TRACKING_EXTENSION_ID
   );
 
-  // TODO: this is a hook that fails if we register/de-register
   if (hasMeasurementTrackingExtension) {
     const contextModule = extensionManager.getModuleEntry(
       '@ohif/extension-measurement-tracking.contextModule.TrackedMeasurementsContext'
     );
 
-    const useTrackedMeasurements = () => useContext(contextModule.context);
-
-    [
-      trackedMeasurements,
-      sendTrackedMeasurementsEvent,
-    ] = useTrackedMeasurements();
+    const tracked = useContext(contextModule.context);
+    trackedMeasurements = tracked?.[0];
+    sendTrackedMeasurementsEvent = tracked?.[1];
+  }
+  if (!sendTrackedMeasurementsEvent) {
+    // if no panels from measurement-tracking extension is used, this code will trun
+    trackedMeasurements = null;
+    sendTrackedMeasurementsEvent = (eventName, { displaySetInstanceUID }) => {
+      MeasurementService.clearMeasurements();
+      hydrateStructuredReport(
+        { servicesManager, extensionManager },
+        displaySetInstanceUID
+      );
+    };
   }
```

Comment:
Can you double check if you really need this

Reply:
Yes, that code in the old version sent a tracked measurement event, which internally called the rehydrate event AFTER showing a dialog asking if it should continue tracking.  In the new code, if it is NOT tracked measurements - that is, only if it is actually running without the tracked measurement panel, it directly calls the rehydrate, which is the code that has to be called to actually load the new measurements into the measurement service.